packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -204,6 +208,7 @@ export function useInnerBlocksProps( props = {}, options = {} ) {
 
 	const blockDropZoneRef = useBlockDropZone( {
 		rootClientId: clientId,
+		dropZoneName: __unstableDropZoneName,
```

Comment:
This will allow the dropzone name to be compared against the draggable origin name during drag and drop events.